pkg/util/util.go

```diff
@@ -298,3 +297,38 @@ func ComponentsPausedForMigration(cluster *corev1.StorageCluster) bool {
 	componentsPaused, err := strconv.ParseBool(cluster.Annotations[constants.AnnotationPauseComponentMigration])
 	return migrating && err == nil && componentsPaused
 }
+
+// HasTopologySpreadConstraintsChanged checks if the deployment has pod topology spread constraints changed
+func HasTopologySpreadConstraintsChanged(
+	nodeTopologyLabelKeys []string,
+	topologySpreadConstraints []v1.TopologySpreadConstraint,
+) bool {
+	nodeTopologyLabelKeySet := make(map[string]bool)
+	constraintTopologyKeySet := make(map[string]bool)
+	for _, key := range nodeTopologyLabelKeys {
+		nodeTopologyLabelKeySet[key] = true
+	}
+	for _, constraint := range topologySpreadConstraints {
+		constraintTopologyKeySet[constraint.TopologyKey] = true
+	}
+	return !reflect.DeepEqual(nodeTopologyLabelKeySet, constraintTopologyKeySet)
```

Comment:
Why check just the keys and not the entire spread constraints? What if the labels have changed?